### DISCUSSION TITLE
Some questions about Argon

### DISCUSSION THREAD

**Discussion by asuaphilie** [argon-rbx/argon#94] [Q&A]
### 1. How do I make argon either ignore instances in studio or sync it with its properties
  I had a ScreenGui in ReplicatedStorage and when I connect argon it synced it to the file system without its properties just the ClassName, next time I connected it reset the properties of my gui

### 2. I can't get argon sourcemap to work properly
  I do `argon sourcemap default.project.json -w -o output.json --async ` and it just outputs it to the console

### 3. Is there any usecase for roblox model files outside of building a plugin?
  I get you might want to use model files for assets to build a plugin but I can't think of anything else that can't be done easier without it
  
### 4. How do people require like this
  ```luau
  require"@Shared/Module"
  ```
  I've seen it done in some docs, maybe it was just an example, but I've seen some code small code snippets doing it too. I think this might be a luau-lsp thing, but while I'm at it ig.

**Comment by DervexDev** [comment]
## 1. How do I make argon either ignore instances in studio or sync it with its properties

To ignore instances take a look at [Syncback Settings](https://argon.wiki/api/project#syncback-settings) that you can specify in your `project.json`.

To sync properties enable the following settings (Roblox Studio plugin):
- Two-Way Sync
- Sync properties

_I know it's misleading but this will be fixed in future updates._

## 2. I can't get argon sourcemap to work properly

Looks like it's a bug, will fix it in `2.0.18`. Temporarily you have to remove the `--async` option.

## 3. Is there any usecase for roblox model files outside of building a plugin?

If you are talking about the `build` command then I listed some other use cases [here](https://argon.wiki/docs/getting-started/common-usage#project-building). If you mean [Roblox Model](https://argon.wiki/docs/getting-started/common-usage#project-building) file type, they are useful to store bulk data e.g. your entire `game.Workspace`.

## 4. How do people require like this

I think it might have been a [lune](https://lune-org.github.io/docs/getting-started/2-introduction/2-built-in-libraries) script, though I'm not sure. Anyway, it's not related to Argon as it currently doesn't touch module resolving.

**Comment by DervexDev** [comment]
> 1. How do I make argon either ignore instances in studio or sync it with its properties

Renamed `Sync properties` option to `Syncback Properties` which no longer requires `Two-Way Sync` setting.

> 2. I can't get argon sourcemap to work properly

Fix available in version [2.0.18](https://github.com/argon-rbx/argon/releases/tag/2.0.18)

@MohamedBannona if you don't have any more questions please mark this discussion as answered. Thanks!
